tests/icon-api.js

```diff
@@ -0,0 +1,187 @@
+import test from 'brittle'
+import RAM from 'random-access-memory'
+import Database from 'better-sqlite3'
+import { drizzle } from 'drizzle-orm/better-sqlite3'
+import { migrate } from 'drizzle-orm/better-sqlite3/migrator'
+import { randomBytes } from 'node:crypto'
+
+import IconApi, { kCreate, kGetIcon } from '../src/icon-api.js'
+import { DataType } from '../src/datatype/index.js'
+import { DataStore } from '../src/datastore/index.js'
+import { createCoreManager } from './helpers/core-manager.js'
+import { iconTable } from '../src/schema/project.js'
+import { IndexWriter } from '../src/index-writer/index.js'
+
+const expectedSmallIcon = randomBytes(128)
+const expectedMediumIcon = randomBytes(128)
+const expectedLargeIcon = randomBytes(128)
+const cm = createCoreManager()
+const sqlite = new Database(':memory:')
+const db = drizzle(sqlite)
+migrate(db, {
+  migrationsFolder: new URL('../drizzle/project', import.meta.url).pathname,
+})
+
+const indexWriter = new IndexWriter({
+  tables: [iconTable],
+  sqlite,
+})
+
+const iconDataStore = new DataStore({
+  namespace: 'config',
+  coreManager: cm,
+  storage: () => new RAM(),
+  batch: async (entries) => indexWriter.batch(entries),
+})
+
+const iconDataType = new DataType({
+  dataStore: iconDataStore,
+  table: iconTable,
+  db,
+})
+
+const iconApi = new IconApi({ iconDataStore, iconDataType })
+
+// eslint-disable-next-line no-unused-vars
+test('icon create and get', async (t) => {
+  const iconDoc = await iconApi[kCreate]({
+    name: 'myIcon',
+    schemaName: 'icon',
+    variants: [
+      {
+        size: 'small',
+        pixelDensity: 1,
+        mimeType: 'image/png',
+        blob: Buffer.from(expectedSmallIcon),
+      },
+    ],
+  })
+
+  const { icon } = await iconApi[kGetIcon]({ iconId: iconDoc.docId })
+  t.alike(icon, expectedSmallIcon)
+})
+
+test(`icon create and fail to find variant with matching mimeType`, async (t) => {
+  const iconDoc = await iconApi[kCreate]({
+    name: 'myIcon',
+    schemaName: 'icon',
+    variants: [
+      {
+        size: 'small',
+        pixelDensity: 1,
+        mimeType: 'image/svg+xml',
+        blob: Buffer.from(expectedSmallIcon),
+      },
+    ],
+  })
+
+  t.exception(
```

Comment:
I don't seem to get how to test an exception. Calling that method throws as expected,  but it doesn't produce a passing test, it just terminates the program. If I wrap it in a try catch, I think it just skips it...

Reply:
This - I think - the only think 'missing' to be able to merge this. 
Test on icon-api could be improved.
i.e. having fixtures for DRY, and maybe exploring cases that I missed??

Reply:
answered this here: https://github.com/digidem/mapeo-core-next/pull/335#discussion_r1372209201